app/assets/stylesheets/base/layout.scss

```diff
@@ -0,0 +1,77 @@
+@import '../vars/variables';
+
+
+// Display
+.hidden {
+  display: none !important;
```

Comment:
are all of these `!important` because we have a CSS code base that's in flux?

Reply:
Actually not - this is part of our frontend approach. We have two parts of frontend: 1. Components & 2. Tailwind.

Components are predefined structures, like cards, modals, etc..

Tailwind lets you create not only views and layouts but it also let you build on top of the components. If you have some component, you can customize it to your needs (if necessary) using tailwind.. For example you have a _form field component_ with label and text input. In your specific scenario you need to make input semitransparent (i don't know why:P). So you would use default component and overwrite part of it with tailwind class `.opacity-50`. So if we ever update default component CSS, your custom component will reflect that change. But you're overwriting existing CSS that may have higher specificity selectors so we need `!important`.

Reply:
Is this the recommended approach with Tailwind? I feel dirty seeing all these `!important`s.

Reply:
Yes and no.

Reply:
[Yes and no. ](https://tailwindcss.com/docs/configuration/#important)

- NO if our frontend was ONLY tailwind (in other words, if we were starting from scratch)
- YES if we have existing frontend codebase and if we also have components defined.

Reply:
OK, I figured it was because of our existing CSS codebase.